docs/RANDOM_SEEDS.md

```diff
@@ -0,0 +1,416 @@
+  pool is not initialized yet.
+
+* Reading from the
+  [`/dev/urandom`](http://man7.org/linux/man-pages/man4/urandom.4.html)
+  pseudo-device will always return random bytes immediately, even if the pool
+  is not initialized. The provided random bytes will be of low quality in this
+  case however. Moreover the kernel will log about all programs using this
+  interface in this state, and which thus potentially rely on an uninitialized
+  entropy pool.
+
+(Strictly speaking there are more APIs, for example `/dev/random`, but these
+should not be used for almost any applications and hence aren't mentioned
+here.)
+
+Note that the time it takes to initialize the random pool may differ between
+systems. If local hardware random number generators are available,
+initialization is likely quick, but particularly in embedded and virtualized
+environments available entropy is small and thus random pool initialization
+might take a long time (up to tens of minutes!).
+
+Modern hardware tends to come with a number of hardware random number
+generators (hwrng), that may be used to relatively quickly fill up the entropy
+pool. Specifically:
+
+* All recent Intel and AMD CPUs provide the CPU opcode
+  [RDRAND](https://en.wikipedia.org/wiki/RdRand) to acquire random bytes. Linux
+  includes random bytes generated this way in its entropy pool, but didn't use
+  to credit entropy for it (i.e. data from this source wasn't considered good
+  enough to consider the entropy pool properly filled even though it was
+  used). This has changed recently however, and most big distributions have
+  turned on the `CONFIG_RANDOM_TRUST_CPU=y` kernel compile time option. This
+  means systems with CPUs supporting this opcode will be able to very quickly
+  reach the "pool filled" state.
+
+* The TPM security chip that is available on all modern desktop systems has a
+  hwrng. It is also fed into the entropy pool, but generally not credited
+  entropy. You may use `rng_core.default_quality=1000` on the kernel command
+  line to change that, but note that this is a global setting affect all
+  hwrngs. (Yeah, that's weird.)
+
+* Many Intel and AMD chipsets have hwrng chips. Their Linux drivers usually
+  don't credit entropy. (But there's `rng_core.default_quality=1000`, see
+  above.)
+
+* Various embedded boards have hwrng chips. Some drivers automatically credit
+  entropy, others do not. Some WiFi chips appear to have hwrng sources too, and
+  they usually do not credit entropy for them.
+
+* `virtio-rng` is used in virtualized environments and retrieves random data
+  from the VM host. It credits full entropy.
+
+* The EFI firmware typically provides a RNG API. When transitioning from UEFI
+  to kernel mode Linux will query some random data through it, and feed it into
+  the pool, but not credit entropy to it. What kind of random source is behind
+  the EFI RNG API is often not entirely clear, but it should generally be some
+  kind of hardware source.
```

Comment:
During the call, I understood that the UEFI standard specifies that this *must* be a hardware source. But looking at the standard, I don't see any such direct quote. Do you have a link to a good source for this requirement? If yes, maybe this part should be reworded.

Reply:
I'll tone this down. I though I had read that in the specs, but can't find it either